File: sdk/typescript/src/types/memory.ts
/** Mirrors memory.service.js's `_toFileDto` — the wire shape for one memory file. */
export interface MemoryFile {
  scope: 'user' | 'agent' | 'workspace';
  /** Set when `scope` is `'agent'` or `'workspace'`. */
  agentId?: string;
  path: string;
  content: string;
  mimeType: string;
  createdAt: string;
  updatedAt: string;
}

/**
 * One agent's group of scoped memory files, as returned by `memory.list()` —
 * used for both `agentMemories` (`scope: 'agent'`) and `agentWorkspaces`
 * (`scope: 'workspace'`).
 */
export interface MemoryAgentGroup {
  agentId: string;
  /** `null` if the Agent no longer exists. */
  agentName: string | null;
  files: MemoryFile[];
}

/** `GET /api/v1/developer/memory` — every memory file for the asserted external user. */
export interface MemoryListResult {
  userFiles: MemoryFile[];
  agentMemories: MemoryAgentGroup[];
  /**
   * Files under an Agent's own `/workspace/` filesystem route — the SAME
   * persistent store `write_file`/`read_file` tool calls under
   * `/workspace/...` actually read and write (deepagents routes that
   * prefix to this Mongo-backed store, NOT the LangGraph checkpoint's
   * `files` state channel — a thread's live/checkpointed state is a
   * different, much narrower thing). Scoped by Agent + Subject, shared
   * across every Thread that Subject has with that Agent.
   */
  agentWorkspaces: MemoryAgentGroup[];
}

export interface MemoryFileScopeParams {
  /** @default 'user' */
  scope?: 'user' | 'agent' | 'workspace';
  /** Required when `scope` is `'agent'` or `'workspace'`. */
  agentId?: string;
}

export interface GetMemoryFileParams extends MemoryFileScopeParams {
  path: string;
}

export interface WriteMemoryFileInput extends MemoryFileScopeParams {
  path: string;
  content: string;
}

export interface DeleteMemoryFileParams extends MemoryFileScopeParams {
  path: string;
}

File: sdk-python/src/personaai/types/memory.py
"""Mirrors ``sdk/src/types/memory.ts``."""

from __future__ import annotations

from typing import Literal, TypedDict


class MemoryFile(TypedDict):
    """One memory file as returned by the API (mirrors ``memory.service.js``'s
    ``_toFileDto``)."""

    scope: Literal["user", "agent"]
    # Set when scope is 'agent'.
    agentId: str | None
    path: str
    content: str
    mimeType: str
    createdAt: str
    updatedAt: str


class MemoryAgentGroup(TypedDict):
    """One agent's group of agent-scoped memory files, as returned by
    ``memory.list()``."""

    agentId: str
    # None if the Agent no longer exists.
    agentName: str | None
    files: list[MemoryFile]


class MemoryListResult(TypedDict):
    """``GET /api/v1/developer/memory`` — every memory file for the asserted
    external user: user-global files plus one group per Agent that has
    agent-scoped memory."""

    userFiles: list[MemoryFile]
    agentMemories: list[MemoryAgentGroup]


class MemoryFileScopeParams(TypedDict, total=False):
    scope: Literal["user", "agent"]  # default: 'user'
    agentId: str  # required when scope is 'agent'


class GetMemoryFileParams(MemoryFileScopeParams):
    path: str


class WriteMemoryFileInput(MemoryFileScopeParams):
    path: str
    content: str


class DeleteMemoryFileParams(MemoryFileScopeParams):
    path: str
